visualization/app/codeCharta/ui/metricChooser/metricChooser.component.ts

```diff
@@ -172,47 +169,32 @@ export class MetricChooserController
 	}
 
 	public onBuildingHovered(data: CodeMapBuildingTransition) {
-		if (data && data.to && data.to.node && data.to.node.attributes) {
-			this.hoveredAreaValue = data.to.node.attributes[this._viewModel.areaMetric]
-			this.hoveredColorValue = data.to.node.attributes[this._viewModel.colorMetric]
-			this.hoveredHeightValue = data.to.node.attributes[this._viewModel.heightMetric]
-
+		if (data && data.to && data.to.node) {
+			this._viewModel.hoveredNode = data.to.node
 			if (data.to.node.deltas) {
-				this.hoveredAreaDelta = data.to.node.deltas[this._viewModel.areaMetric]
-				this.hoveredColorDelta = data.to.node.deltas[this._viewModel.colorMetric]
-				this.hoveredHeightDelta = data.to.node.deltas[this._viewModel.heightMetric]
-
-				this.hoveredDeltaColor = this.getHoveredDeltaColor()
-			} else {
-				this.hoveredAreaDelta = null
-				this.hoveredColorDelta = null
-				this.hoveredHeightDelta = null
-				this.hoveredDeltaColor = null
+				this._viewModel.deltaColor = this.getHoveredDeltaColor()
 			}
 		} else {
-			this.hoveredAreaValue = null
-			this.hoveredColorValue = null
-			this.hoveredHeightValue = null
-			this.hoveredHeightDelta = null
-			this.hoveredAreaDelta = null
-			this.hoveredColorDelta = null
+			this._viewModel.hoveredNode = null
 		}
+		this.synchronizeAngularTwoWayBinding()
 	}
 
 	private getHoveredDeltaColor() {
-		let colors = {
-			0: "green",
-			1: "red"
-		}
+		const heightDelta: number = this._viewModel.hoveredNode.deltas[this._viewModel.heightMetric]
 
-		if (this.hoveredHeightDelta > 0) {
-			return colors[Number(this.settingsService.getSettings().appSettings.invertDeltaColors)]
-		} else if (this.hoveredHeightDelta < 0) {
-			return colors[Number(!this.settingsService.getSettings().appSettings.invertDeltaColors)]
+		if (heightDelta > 0) {
+			return "green"
+		} else if (heightDelta < 0) {
+			return "red"
 		} else {
 			return "inherit"
 		}
 	}
+
+	private synchronizeAngularTwoWayBinding() {
+		this.$timeout(() => {})
```

Comment:
What is this doing?

Reply:
you will find this function `synchronizeAngularTwoWayBinding` in some components. The angular lifecycle handles the twoWayBinding of data between the controller and the html template. This means angular decides, when it is a good moment to update the html data. For some reason, angular does not always want update the data, especially after events are received and data was changed.

The timeout function (with 0 seconds by default) is a workaround, which indirectly triggers updating the html data.